tests/integration/devfile/cmd_delete_test.go

```diff
@@ -0,0 +1,90 @@
+package devfile
+
+import (
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/redhat-developer/odo/tests/helper"
+)
+
+var _ = Describe("odo delete command tests", func() {
+	var cmpName string
+	var commonVar helper.CommonVar
+
+	// This is run before every Spec (It)
+	var _ = BeforeEach(func() {
+		commonVar = helper.CommonBeforeEach()
+		cmpName = helper.RandString(6)
+		helper.Chdir(commonVar.Context)
+	})
+
+	// This is run after every Spec (It)
+	var _ = AfterEach(func() {
+		helper.CommonAfterEach(commonVar)
+	})
+
+	When("a component is bootstrapped", func() {
+		BeforeEach(func() {
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
+			helper.Cmd("odo", "project", "set", commonVar.Project).ShouldPass()
+			helper.Cmd("odo", "init", "--name", cmpName, "--devfile-path", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile-deploy.yaml")).ShouldPass()
+		})
+		When("the component is deployed in DEV mode", func() {
+			BeforeEach(func() {
+				session := helper.CmdRunner("odo", "dev")
+				defer session.Kill()
+				helper.WaitForOutputToContain("Waiting for something to change", 180, 10, session)
+
+				list := commonVar.CliRunner.Run("get", "deployment", "-n", commonVar.Project).Out.Contents()
+				Expect(list).To(ContainSubstring(cmpName))
+			})
+
+			When("the component is deleted using its name and namespace", func() {
+				var out string
+				BeforeEach(func() {
+					out = helper.Cmd("odo", "delete", "component", "--name", cmpName, "--namespace", commonVar.Project, "-f").ShouldPass().Out()
```

Comment:
Perhaps you should move out of the `commonVar.Context` so that you ensure that you do not have access to the `devfile.yaml`